phoenix-core/src/main/java/org/apache/phoenix/compile/JoinCompiler.java

```diff
@@ -438,9 +471,46 @@ public Expression compilePostFilterExpression(StatementContext context) throws S
             return vector;
         }
 
-        public JoinTable getSubJoinTableWithoutPostFilters() {
-            return joinSpecs.size() > 1 ? new JoinTable(table, joinSpecs.subList(0, joinSpecs.size() - 1)) :
-                new JoinTable(table);
+        /**
```

Comment:
Added a new JoinTable.createSubJoinTable method to replace the old JoinTable.getSubJoinTableWithoutPostFilters.
In addition to create the new JoinTable, JoinTable.createSubJoinTable also pushs down the PostFilter to the new created JoinTable .